servers/jax-rs-tests/src/main/java/org/projectnessie/jaxrs/gc/AbstractRestGCTest.java

```diff
@@ -0,0 +1,427 @@
+    fillExpectedContents(branch1, 4, expectedExpired);
+
+    final Instant cutoffTime = Instant.now();
+
+    // two commits for "table_1" on branch1
+    table1 =
+        commitSingleOp(
+            prefix, branch1, table1.hash, 44, "cid_1", "table_1", "file3", table1.content, null);
+    table1 =
+        commitSingleOp(
+            prefix, branch1, table1.hash, 45, "cid_1", "table_1", "file4", table1.content, null);
+    // one commit for "table_2" on branch1
+    table2 =
+        commitSingleOp(
+            prefix, branch1, table2.hash, 44, "cid_2", "table_2", "file3", table2.content, null);
+
+    performGc(
+        cutoffTime,
+        null,
+        expectedExpired,
+        new HashSet<>(Arrays.asList(prefix + "_cid_1", prefix + "_cid_2")),
+        true);
+
+    // test GC with commit protection. No commits should be expired.
+    performGc(
+        cutoffTime,
+        null,
+        Collections.emptyMap(),
+        new HashSet<>(Arrays.asList(prefix + "_cid_1", prefix + "_cid_2")),
+        false);
+  }
+
+  @Test
+  public void testSingleRefDropTable() throws BaseNessieClientServerException {
+    String prefix = "singleRefDropTable";
+    Map<String, Set<Content>> expectedExpired = new HashMap<>();
+
+    Branch branch1 = createBranch(prefix);
+    // two commits for "table_1" on branch1
+    CommitOutput table1 =
+        commitSingleOp(
+            prefix, branch1, branch1.getHash(), 42, "cid_1", "table_1", "file1", null, null);
+    table1 =
+        commitSingleOp(
+            prefix, branch1, table1.hash, 43, "cid_1", "table_1", "file2", table1.content, null);
+    // two commits for "table_2" on branch1
+    CommitOutput table2 =
+        commitSingleOp(
+            prefix, branch1, branch1.getHash(), 42, "cid_2", "table_2", "file1", null, null);
+    table2 =
+        commitSingleOp(
+            prefix, branch1, table2.hash, 43, "cid_2", "table_2", "file2", table2.content, null);
+    // All 4 commits are expected to be expired as it is before the cutoff time and there are some
+    // more commits for
+    // each table after cutoff timestamp (i,e these 4 commits are not head commits)
+    fillExpectedContents(branch1, 4, expectedExpired);
+
+    final Instant cutoffTime = Instant.now();
+
+    // two commit for "table_1" on branch1
+    table1 =
+        commitSingleOp(
+            prefix, branch1, table1.hash, 44, "cid_1", "table_1", "file3", table1.content, null);
+    table1 =
+        commitSingleOp(
+            prefix, branch1, table1.hash, 45, "cid_1", "table_1", "file4", table1.content, null);
+
+    // one commit for "table_2" on branch1
+    table2 =
+        commitSingleOp(
+            prefix, branch1, table2.hash, 44, "cid_2", "table_2", "file3", table2.content, null);
+    // drop table "table_2"
+    dropTableCommit(prefix, branch1, table2.hash, "table_2");
+    // table_2's one commit (after cutoff timestamp) is also expected to be expired as table is
```

Comment:
I suspect, this is wrong.
The commit created in line 124 is after the cut-off time and should be considered "live" / usable via time-travel.